BUILD

```diff
@@ -1,21 +1,22 @@
+package(default_visibility = ["//visibility:public"])
```

Comment:
Would it be possible somehow to have the bazel build files all in one folder, `bazel_rules` or something? There are now be files littered across `rules` and the root of the project and it's a bit messy to know which build system each belongs to.

Reply:
I can, there are trade-offs either way. The WORKSPACE file in the root is not strictly needed, so that could be removed altogether. The BUILD file in the root can also be removed, however it comes at a cost in ease of use of use when referencing this project remotely. In other words removing the BUILD file in the root the requires the build file to be specified in the users WORKSPACE file. e.g.
This:
```py
git_repository(
    name = "com_github_nanopb_nanopb",
    commit = "766ee1cebf6e1500dd9bd94fc217abb554ef55e7",
    remote = "https://github.com/silvergasp/nanopb.git",
)
```
Would become this:
```py
new_git_repository(
    name = "com_github_nanopb_nanopb",
    commit = "766ee1cebf6e1500dd9bd94fc217abb554ef55e7",
    remote = "https://github.com/silvergasp/nanopb.git",
    build_file_content = """
package(default_visibility = ["//visibility:public"])

licenses(["notice"])

exports_files(["LICENSE.txt"])

cc_library(
    name = "nanopb",
    srcs = [
        "pb_common.c",
        "pb_decode.c",
        "pb_encode.c",
    ],
    hdrs = [
        "pb.h",
        "pb_common.h",
        "pb_decode.h",
        "pb_encode.h",
    ],
    strip_include_prefix = ".",
    visibility = ["//visibility:public"],
)
"""
)
```
The other alternative is to merge everything in the /rules directory, into a single /rules.bzl file, and then /rules/BUILD can be merged with /BUILD.

Reply:
So if I understand correctly, the `/BUILD` is now for building the C files, while `rules/BUILD` is for running the generator?

`/BUILD` and `/WORKSPACE` have been in place for long enough that other projects probably depend on them, so it's probably not a good idea to move them. If the other rules can be merged into `/BUILD`, that sounds good to me. It would also be great if you can add a comment on top of each of the files, something like "This file is for the bazel build system, see examples/bazel_simple" and if there is any useful extra info about the purpose of a particular file.